src/irr/asset/COBJMeshFileLoader.cpp

```diff
@@ -487,359 +551,121 @@ asset::SAssetBundle COBJMeshFileLoader::loadAsset(io::IReadFile* _file, const as
+        for (uint32_t i = CMTLPipelineMetadata::SMtl::EMP_REFL_POSX + 1u; i < CMTLPipelineMetadata::SMtl::EMP_REFL_POSX + 6u; ++i)
+        {
+            images[i] = nullptr;
+        }
+    }
 
-				currMaterial->Material.DiffuseColor.setAlpha( (int32_t)(dValue * 255) );
-                //TODO
-				//if (dValue<1.0f)
-				//	currMaterial->Material.MaterialType = video::EMT_TRANSPARENT_VERTEX_ALPHA;
-			}
-			break;
-			case 'T':
-			if ( currMaterial )
-			{
-				switch ( bufPtr[1] )
-				{
-				case 'f':		// Tf - Transmitivity
-					const uint32_t COLOR_BUFFER_LENGTH = 16;
-					char redStr[COLOR_BUFFER_LENGTH];
-					char greenStr[COLOR_BUFFER_LENGTH];
-					char blueStr[COLOR_BUFFER_LENGTH];
-
-					bufPtr = goAndCopyNextWord(redStr,   bufPtr, COLOR_BUFFER_LENGTH, bufEnd);
-					bufPtr = goAndCopyNextWord(greenStr, bufPtr, COLOR_BUFFER_LENGTH, bufEnd);
-					bufPtr = goAndCopyNextWord(blueStr,  bufPtr, COLOR_BUFFER_LENGTH, bufEnd);
-
-					float red,green,blue;
-					sscanf(redStr,"%f",&red);
-					sscanf(greenStr,"%f",&green);
-					sscanf(blueStr,"%f",&blue);
-					float transparency = ( red+green+blue ) / 3;
-
-					currMaterial->Material.DiffuseColor.setAlpha( (int32_t)(transparency * 255) );
-                    //TODO
-					//if (transparency < 1.0f)
-					//	currMaterial->Material.MaterialType = video::EMT_TRANSPARENT_VERTEX_ALPHA;
-				}
-			}
-			break;
-#endif
-			default: // comments or not recognised
-			break;
-		} // end switch(bufPtr[0])
-		// go to next line
-		bufPtr = goNextLine(bufPtr, bufEnd);
-	}	// end while (bufPtr)
+    return images;
+}
+
+core::smart_refctd_ptr<ICPUDescriptorSet> COBJMeshFileLoader::makeDescSet(const images_set_t& _images, ICPUDescriptorSetLayout* _dsLayout)
+{
+    auto ds = core::make_smart_refctd_ptr<asset::ICPUDescriptorSet>(
+        core::smart_refctd_ptr<ICPUDescriptorSetLayout>(_dsLayout)
+    );
+    for (uint32_t i = 0u, d = 0u; i <= CMTLPipelineMetadata::SMtl::EMP_REFL_POSX; ++i)
+    {
+        if (!_images[i])
+            continue;
 
-	// end of file. if there's an existing material, store it
-	if ( currMaterial )
-		_ctx.Materials.push_back( currMaterial );
+        constexpr IImageView<ICPUImage>::E_TYPE viewType[2]{ IImageView<ICPUImage>::ET_2D, IImageView<ICPUImage>::ET_CUBE_MAP };
+        constexpr uint32_t layerCount[2]{ 1u, 6u };
+
+        const bool isCubemap = (i == CMTLPipelineMetadata::SMtl::EMP_REFL_POSX);
+
+        ICPUImageView::SCreationParams viewParams;
+        viewParams.flags = static_cast<ICPUImageView::E_CREATE_FLAGS>(0u);
+        viewParams.format = _images[i]->getCreationParameters().format;
+        viewParams.image = _images[i];
+        viewParams.viewType = viewType[isCubemap];
+        viewParams.subresourceRange.baseArrayLayer = 0u;
+        viewParams.subresourceRange.layerCount = layerCount[isCubemap];
+        viewParams.subresourceRange.baseMipLevel = 0u;
+        viewParams.subresourceRange.levelCount = 1u;
```

Comment:
patch our default CPU to GPU object converter so that it always extends the mipmap chain to the full max value and computes the missing mipmaps (you can hack it with a blit for now or a compute shader mip-mapping stub)

Reply:
obvs don't touch the loaders and CPU Images ;)

Reply:
ImageView CPU2GPU will also need the `levelCount` patch-up